# to go backward pass False
bool forward
---
bool success
string message
